pkg/resource/generator.go

```diff
@@ -86,8 +96,132 @@ func (g *Generator) list(cr *regopapi.ImageRegistry) ([]Mutator, error) {
 	return mutators, nil
 }
 
+// syncStorage checks:
+// 1.)  to make sure that an existing storage medium still exists and we can access it
+// 2.)  to see if the storage medium name changed and we need to:
+//      a.) check to make sure that we can access the storage or
+//      b.) see if we need to try to create the new storage
+func (g *Generator) syncStorage(cr *regopapi.ImageRegistry, modified *bool) error {
```

Comment:
We should only be managing the storage in one place, so the fact that you're doing this in both CompleteConfiguration and syncStorage is wrong to me.

Eventually consistent behavior would say the CompleteConfiguration itself should not be responsible for creating/managing the storage.  It can setup whatever initial configuration is needed, but then the standard sync/watch loop should see the configuration (either as created by CompleteConfiguration, or edited/provided by the user) and make the necessary storage checks/updates.

As soon as you start doing it from two different code paths you're creating assumptions about system state and initialization order.

Reply:
Updated in next push